src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/DictionaryModelBinder.cs

```diff
@@ -22,12 +24,25 @@ public class DictionaryModelBinder<TKey, TValue> : CollectionModelBinder<KeyValu
         private readonly IModelBinder _valueBinder;
 
         /// <summary>
-        /// Creates a new <see cref="DictionaryModelBinder{TKey, TValue}"/>.
+        /// <para>This constructor is obsolete and will be removed in a future version.</para>
```

Comment:
Need `The recommended alternative` verbiage here and in the `[Obsolete(...)]` text